cmd/gen-key/ecdsa.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/miekg/pkcs11"
+)
+
+var stringToCurve = map[string]elliptic.Curve{
+	"P-224": elliptic.P224(),
+	"P-256": elliptic.P256(),
+	"P-384": elliptic.P384(),
+	"P-521": elliptic.P521(),
+}
+
+var curveToOID = map[elliptic.Curve][]byte{
+	elliptic.P224(): []byte{6, 5, 43, 129, 4, 0, 33},
```

Comment:
Do you have a reference for this? They return a static `elliptic.Curve` struct which should never change in such a way that would break this usage. Given `elliptic.Curve` is used as a argument to a `ecdsa.GenerateKey` and a field value on `ecdsa.PublicKey` it is bound by the  golang 1 backwards compatibility guarantee.

Reply:
> They return a static elliptic.Curve struct which should never change in such a way that would break this usage.

`elliptic.Curve` is not a struct, it's an interface. There are no guarantees about what the underlying thing is. I realize that under the current implementation these always return the same, but I'd like to see documentation to that effect before we rely on it. Want to file an issue on the Go repo asking for such documentation, or alternatively documentation on the best way to index curves in a map?